### PR TITLE
Refactor Passive Component Checks

### PR DESCRIPTION
**Pull request by m-hilgendorf**

### CODE REVIEW

File: utils/passive-checks/capacitor-checks.stanza [hunk 1]
```diff
@@ -0,0 +1,315 @@
+#use-added-syntax(jitx)
```
Comment (m-hilgendorf) [inline utils/passive-checks/capacitor-checks.stanza:1]: Keep around the old code for posterity

File: utils/checks.stanza [hunk 2]
```diff
@@ -54,14 +57,15 @@ public defn erc-check-design (module:JITXObject):
   inside pcb-module:
     for i in populated-components(module) do :
       check rated-temperature(i)
-      check capacitor-component(i) when has-property?(i.capacitor)
-      check resistor-component(i)  when has-property?(i.resistor)
-      check inductor-component(i)  when has-property?(i.inductor)
-      check diode-component(i)     when has-property?(i.diode)
```
Comment (m-hilgendorf) [inline utils/checks.stanza:60]: This is not an accident: the diode component checks are redundant (only pin connectivity is checked, which is handled elsewhere)

File: utils/checks.stanza [hunk 3]
```diff
@@ -54,14 +57,15 @@ public defn erc-check-design (module:JITXObject):
...
+      
+      check-capacitor(i) when has-property?(i.capacitor)
+      check-resistor(i)  when has-property?(i.resistor)
+      check-inductor(i)  when has-property?(i.inductor)
```
Comment (m-hilgendorf) [inline utils/checks.stanza:63]: Implementation moved to `passive-checks/*`

File: utils/passive-checks/capacitor-checks.stanza [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+  import ocdb/utils/design-vars
+  import ocdb/space-derating
+  import ocdb/utils/generator-utils
+  import ocdb/utils/property-structs
+  import ocdb/utils/tolerance
+  import ocdb/utils/passive-checks/utils
+
+;==============================================================================
+;========================== Capacitor Checks ==================================
+;==============================================================================
```
Comment (m-hilgendorf) [inline utils/passive-checks/capacitor-checks.stanza:16]: This is the best example of the new checks architecture.

File: utils/checks.stanza (hunk 2 above)
Comment (bhusang) [reply utils/checks.stanza:60]: We should add 1-2 more diode checks (like reverse breakdown voltage, max forward current) but your simplification is ok. Let's plan to add diodes to the passive-checks.

**Review by bhusang — APPROVED** [review]
The rewrite looks great. I like how you have repartitioned the checks into separate files with a clear distinction on finding the properties needed. Going forward this will be a great base from which to evolve the checks comprehensively.

File: derating/space-derating.stanza [hunk 5]
```diff
@@ -5,31 +5,31 @@ defpackage ocdb/space-derating :
 public var DESIGN-TEMPERATURE:Double = 0.0
 
 ; RESISTOR DERATING PARAMETERS
-public val DERATE-RESISTOR-POWER = 0.6
-public val DERATE-INDUCTOR-POWER = 0.6
+public var DERATE-RESISTOR-POWER = 0.6
+public var DERATE-INDUCTOR-POWER = 0.6
```
Comment (m-hilgendorf) [inline derating/space-derating.stanza:9]: SIdenote: these are now `var` in case a user wishes to customize deratings in a design.

**Pull request merged**